src/AccessionTables/components/LocationSelection/LocationSelection.js

```diff
@@ -5,26 +5,23 @@ import { useIntl } from 'react-intl';
 
 import { Selection } from '@folio/stripes-acq-components';
 
-import { Location } from '../Location';
-
 // todo: this is forked from stripes-smart-components, got to figure out the way to merge
 
-// todo: does not highlights the search criterion in the results
 const filter = (value, data) => {
-  return data.filter(o => new RegExp(escapeRegExp(value), 'i').test(o.label.props?.location.name));
+  return data.filter(o => new RegExp(escapeRegExp(value), 'i').test(o.label));
 };
 
-
 export const LocationSelection = ({ locations, placeholder, ...rest }) => {
   const { formatMessage } = useIntl();
 
   const finalPlaceholder = placeholder || formatMessage({ id: 'stripes-smart-components.ls.locationPlaceholder' });
 
   const locationOpts = locations.map(loc => ({
-    label: <Location location={loc} />,
+    label: `${loc.name} (${loc.code})`,
```

Comment:
That's a pretty simple fix, which is great in a way.

But I would propose to consider maybe a little more complicated approach I took in #111
The goal is to keep the DRY `<Location>` presentation, and still be able to use the highlighting inside.

Reply:
It seems to me that this complexity is not worth the DRY rule in this case. It's harder to test, harder to read and harder to debug. Let's wait for the opinion of other reviewers, because i could be wrong

Reply:
The thing is we've got similar problem related to our code, to using React node as a label in `stripes-smart-components`.
It was recently fixed by @mkuklis [here](https://github.com/folio-org/stripes-smart-components/pull/1084/files#diff-e7080135ea21fa0f0725156616117122b3c9e44467607528c83bac3a17730d37R43-R46) the same way you do it.

But I've got an idea of separating the Remote Storage related code from other modules. For now it's hard-wired there, and it's not a good architecture, too much coupling.  I've made a [ticket](https://issues.folio.org/browse/UIRS-58) for that.

In case of `stripes-smart-components` my thought was to add "remote" badge to the locations as a stripes plugin.

But for that we need to be able to deal with option label in form of React element, not just the string, I believe.

Reply:
I think you both make good points. IMO it rests on that UIRS-58 ticket you filed, @axelhunn . If that's gonna get handled in the next month or two, then I wouldn't be too concerned with adding the extra complexity now when it's inevitably gonna get refactored. So I'd leave it simpler as @vashjs has it currently.

But if it's just a nice-to-have and you don't reasonably see it actually being done soon - whether that's formally allocated or in any extra time you find - then I do think it probably makes more sense to keep using the `Location` component like in #111 so that the various implementations of location-lookup code in the repo don't fragment and we lose of track when that refactor work _is_ done.

If you were to pin me down...I'd say take the complex approach unless you are absolutely certain you're gonna find time to do the rework.

Reply:
I agree with @doytch. I think this PR is minimal and better than what we had before so I would just go ahead and merge. When https://issues.folio.org/browse/UIRS-58 is done this can be revisited / refactored.

Reply:
Yes, on second thought I'm inclined to make it a part of https://issues.folio.org/browse/UIRS-58 solution - that would be more consistent.

Reply:
@doytch The rework should be done anyway IMO. This whole `LocationSelection` is a fork from `stripes-smart-components`, and it's got to be un-forked. But when this will be prioritized - I don't really know.